ui/src/app/components/side-nav.spec.tsx

```diff
@@ -0,0 +1,39 @@
+import {mount} from 'enzyme';
+import * as React from 'react';
+
+import {SideNav, SideNavItem, SideNavProps} from './side-nav';
+import {ProfileStubVariables} from "../../testing/stubs/profile-api-stub";
+import {DataAccessLevel} from "../../generated/fetch";
+
+
+describe('SideNav', () => {
+  const existingNames = [];
+  const props: SideNavProps = {
+    profile: ProfileStubVariables.PROFILE_STUB,
+    bannerAdminActive: false,
+    workspaceAdminActive: false,
+    homeActive: false,
+    libraryActive: false,
+    onToggleSideNav: () => {},
+    profileActive: false,
+    userAdminActive: false,
+    workspacesActive: false,
+  };
+  const component = () => mount(<SideNav {...props}/>);
+  it('should render', () => {
+    const wrapper = component();
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('disables options when user not registered', () => {
+    const wrapper = mount(<SideNav {...props} profile={{...ProfileStubVariables.PROFILE_STUB,
+      dataAccessLevel: DataAccessLevel.Unregistered}}/>);
+    const disabledItems = ['Your Workspaces', 'Featured Workspaces', 'User Support'];
+    wrapper.find(SideNavItem).forEach(sideNavItem => {
+      if (disabledItems.includes(sideNavItem.text())) {
```

Comment:
It would be better if this test also asserted that you found all the disabled items you were looking for. e.g. by removing elements from disabledItems as you find them and then asserting it's an empty list at the end. Or maybe by keeping a counter.

For example, if I added the string "asdf" to disabledItems above, the test wouldn't fail (more realistic scenario: we change the label text for one of these buttons and forget to update tests).